src/lib/HsToCoq/ConvertHaskell/Module.hs

```diff
@@ -103,9 +103,12 @@ convertHsGroup HsGroup{..} = do
                                                             (cdef^.convDefType)
                                                             (cdef^.convDefBody) NotExistingClass
                              in pure $
-                                [ if useProgram
-                                  then ProgramSentence (DefinitionSentence def) obl
-                                  else DefinitionSentence def ] ++
+                                [ case (cdef^.convDefBody) of
+                                    Fix (FixOne (FixBody qual bind ord mterm term))
+                                      -> FixpointSentence (Fixpoint [(FixBody qual (fromList $ (cdef^.convDefArgs) ++ (toList (bind))) ord mterm term)] [])
+                                    _ -> if useProgram
+                                         then ProgramSentence (DefinitionSentence def) obl
+                                         else DefinitionSentence def ] ++
```

Comment:
A couple minor comments:

1. It seems to me that `useFixpoint` should be on the outside here – right now, this looks like it will block `Program` from being applied to `fix`points/`Fixpoint`s.  But that might be applied separately, so I'm not 100% sure.

2. I think we have some code that does a similar job of pulling parameters out of the `convDefArgs` and into the outer bindings.  Perhaps that can be reused?  For instance, I had to convince myself that this put the arguments in the right order, so abstracting that out would be handy.  See `toProgramFixpointSentence` and `fixConnComp`.

3. In general, we should avoid `NEL.fromList`, so we have `(<++) :: Foldable f => f a -> NonEmpty a -> NonEmpty a` and similarly for `(++>)`.